pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/SegmentDeletionManager.java

```diff
@@ -132,25 +134,36 @@ private synchronized void deleteSegmentFromPropertyStoreAndLocal(String tableNam
         return;
       }
 
-      switch (TableNameBuilder.getTableTypeFromTableName(tableName)) {
-        case OFFLINE:
-          if (_localDiskDir != null) {
-            File fileToDelete = new File(new File(_localDiskDir, tableName), segmentId);
-            if (fileToDelete.exists()) {
-              FileUtils.deleteQuietly(fileToDelete);
-              LOGGER.info("Delete segment : " + segmentId + " from local directory : " + fileToDelete.getAbsolutePath());
-            } else {
-              LOGGER.warn("Not found local segment file for segment : " + segmentId);
-            }
-          } else {
-            LOGGER.info("localDiskDir is not configured, won't delete anything from disk");
+      final String rawTableName = TableNameBuilder.extractRawTableName(tableName);
+      if (_localDiskDir != null) {
+        File fileToMove = new File(new File(_localDiskDir, rawTableName), segmentId);
+        if (fileToMove.exists()) {
+          File targetDir = new File(new File(_localDiskDir, DELETED_SEGMENTS), rawTableName);
+          try {
+            FileUtils.moveFileToDirectory(fileToMove, targetDir, true);
```

Comment:
Since we're not actually deleting the segments, we will eventually fill up the disk, no? Do we want to have some code in Pinot that will clean up this directory on an ongoing basis?